src/test/java/org/folio/circulation/services/LogCheckInServiceTest.java

```diff
@@ -0,0 +1,132 @@
+package org.folio.circulation.services;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.awaitility.Awaitility;
+import org.folio.circulation.domain.CheckInProcessRecords;
+import org.folio.circulation.domain.Item;
+import org.folio.circulation.domain.representations.CheckInByBarcodeRequest;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.CollectionResourceClient;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.ServerErrorFailure;
+import org.folio.circulation.support.http.client.Response;
+import org.joda.time.DateTime;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.stubbing.Answer;
+
+import io.vertx.core.json.JsonObject;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LogCheckInServiceTest {
+  private static final int CHECK_IN_SERVER_TIMEOUT = 200;
+
+  @Mock
+  private Clients clients;
+  @Mock
+  private CollectionResourceClient checkInStorageClient;
+  private LogCheckInService logCheckInService;
+  private volatile Result<Response> spyLastResult;
+
+  @Before
+  public void setUp() {
+    when(clients.checkInStorageClient())
+      .thenReturn(checkInStorageClient);
+    logCheckInService = new LogCheckInService(clients);
+  }
+
+  @Test
+  public void logCheckInOperationIsNotBlocked() {
+    final CheckInProcessRecords context = checkInProcessRecords();
+
+    when(checkInStorageClient.post(any(JsonObject.class)))
+      .thenAnswer(checkInProcessTakesLongTime());
+
+    final CompletableFuture<Result<CheckInProcessRecords>> logCheckInOperation =
+      logCheckInService.logCheckInOperation(context);
+
+    final Result<CheckInProcessRecords> logResult = logCheckInOperation
+      .getNow(Result.failed(new ServerErrorFailure("Uncompleted")));
+
+    assertThat(logResult.succeeded(), is(true));
+    assertThat(logResult.value(), is(context));
+
+    Awaitility.await()
+      .atMost(500, TimeUnit.MILLISECONDS)
+      .until(() -> spyLastResult != null && spyLastResult.value() != null);
+
+    verify(spyLastResult, times(1)).failed();
+    // Verify the logging is not executed for successful result.
+    verify(spyLastResult, never()).cause();
+  }
+
+  @Test
+  public void logCheckInOperationFailureDoNotStopCheckInFlow() {
```

Comment:
I think this needs discussing with Emma. Personally, I think that failing to record that a check in has occurred should fail the check in process entirely.